RevenueCat/Scripts/PresentedOfferingContext.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using RevenueCat.SimpleJSON;
+
+public partial class Purchases
+{
+    /// <summary>
+    /// Stores context about the offering a package was presented from.
+    /// </summary>
+    public class PresentedOfferingContext
+    {
+        public readonly string OfferingIdentifier;
+        [CanBeNull] public readonly string PlacementIdentifier;
+        [CanBeNull] public readonly PresentedOfferingTargetingContext TargetingContext;
+
+        public PresentedOfferingContext(JSONNode response)
+        {
+            OfferingIdentifier = response["offeringIdentifier"];
+            PlacementIdentifier = response["placementIdentifier"];
+            TargetingContext = new PresentedOfferingTargetingContext(response["targetingContext"]);
+        }
+
+        public override string ToString()
+        {
+            return $"{nameof(OfferingIdentifier)}: {OfferingIdentifier}\n" +
+                   $"{nameof(PlacementIdentifier)}: {PlacementIdentifier}\n" + 
+                   $"{nameof(TargetingContext)}: {TargetingContext}";
+        }
+
+        public string ToJson()
+        {
+            var contextDict = new JSONObject();
+            contextDict["offeringIdentifier"] = OfferingIdentifier;
+            if (PlacementIdentifier != null) {
+                contextDict["placementIdentifier"] = PlacementIdentifier;
+            }
+
+            if (TargetingContext != null) {
+                var targetingDict = new JSONObject();
+                targetingDict["revision"] = TargetingContext.Revision;
+                targetingDict["ruleId"] = TargetingContext.RuleId;
+
+                contextDict["targetingContext"] = targetingDict;
+            }
+            
+            return contextDict.ToString();
+        }
```

Comment:
Unity cannot nicely serialize this object to JSON because of some properties being nullable. This method is a manual workaround for that 🤷‍♂️